### PR TITLE
Optimize distributor tests

### LINKED ISSUE #2557: Distributor tests take more than a minute to run
#### Describe the issue

Tests from `./pkg/distributor` take more than 1 minute to run, which is too much for unit tests.

```
/w/github.com/grafana/mimir on  main 11:15:23
$ go test -count=1 ./pkg/distributor/...
ok  	github.com/grafana/mimir/pkg/distributor	85.809s
?   	github.com/grafana/mimir/pkg/distributor/distributorpb	[no test files]
ok  	github.com/grafana/mimir/pkg/distributor/forwarding	0.013s

~/w/github.com/grafana/mimir on  main 11:16:51
$ go test -count=1 ./pkg/distributor/...
ok  	github.com/grafana/mimir/pkg/distributor	85.482s
?   	github.com/grafana/mimir/pkg/distributor/distributorpb	[no test files]
ok  	github.com/grafana/mimir/pkg/distributor/forwarding	0.013s
```

#### To Reproduce

`go test -count=1 ./pkg/distributor/...`

#### Expected behavior

It should take less.

#### Environment

Happens on my native golang installation and also when running `make test` (running through the mimir build image).

### PR DESCRIPTION
**Pull request by ravilushqa**
#### What this PR does
Decrease tests execution time

Results in my machine:
Before:
```bash
go test -count=1 ./pkg/distributor/...
ok      github.com/grafana/mimir/pkg/distributor        91.661s
?       github.com/grafana/mimir/pkg/distributor/distributorpb  [no test files]
ok      github.com/grafana/mimir/pkg/distributor/forwarding     0.470s
```
After:
```bash
go test -count=1 ./pkg/distributor/... 
ok      github.com/grafana/mimir/pkg/distributor        34.071s
?       github.com/grafana/mimir/pkg/distributor/distributorpb  [no test files]
ok      github.com/grafana/mimir/pkg/distributor/forwarding     0.171s
```

#### Which issue(s) this PR fixes or relates to

Fixes #2557

#### Checklist

- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by 56quarters** [comment]
Thanks for this contribution! Instead of spawning goroutines here, can the `t.Parallel()` method be called from inside the test?

**Comment by ravilushqa** [comment]
> Thanks for this contribution! Instead of spawning goroutines here, can the `t.Parallel()` method be called from inside the test?

Probably yes. But the result will be slower
TestDistributor_PushQuery execution:
before changes: 54sec 
t.Parallel(): 7sec
concurrent: 500ms

**Comment by 56quarters** [comment]
> > Thanks for this contribution! Instead of spawning goroutines here, can the `t.Parallel()` method be called from inside the test?
> 
> Probably yes. But the result will be slower TestDistributor_PushQuery execution: before changes: near 1m
> 

I think that's fine and still an improvement. I'd prefer to use something built into the Go test module instead of spawning manually starting goroutines.

**Comment by ravilushqa** [comment]
> I think that's fine and still an improvement. I'd prefer to use something built into the Go test module instead of manually starting goroutines.

Make sense, thanks for the review. Changed to `t.Parallel()`

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ### Grafana Mimir
 
+* [ENHANCEMENT] Distributor: Decreased distributor tests execution time. #2557
```
Comment (56quarters) [inline CHANGELOG.md:7]: Can you group this entry with the other `ENHANCEMENT` entries?

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,7 @@
 * [FEATURE] Querier: enabled support for queries with negative offsets, which are not cached in the query results cache. #2429
 * [FEATURE] EXPERIMENTAL: OpenTelemetry Metrics ingestion path on `/otlp/v1/metrics`. #695 #2436
 * [FEATURE] Querier: Added support for tenant federation to metric metadata endpoint. #2467
+* [ENHANCEMENT] Distributor: Decreased distributor tests execution time. #2557
```
Comment (56quarters) [inline CHANGELOG.md:25]:
```suggestion
* [ENHANCEMENT] Distributor: Decreased distributor tests execution time. #2562
```

**Review by 56quarters — APPROVED** [review]
Thanks!

File: pkg/distributor/distributor_test.go [hunk 3]
```diff
@@ -973,6 +973,7 @@ func TestDistributor_PushQuery(t *testing.T) {
 
 	for _, tc := range testcases {
 		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
```
Comment (pracucci) [inline pkg/distributor/distributor_test.go:976]: I don't think it's correct. We need to change `tc` variable scope.

Comment (pracucci) [reply pkg/distributor/distributor_test.go:976]: See: https://github.com/grafana/mimir/pull/2567

**Pull request merged**